pkg/rancher-desktop/config/settings.ts

```diff
@@ -474,114 +473,142 @@ function updateSettings(settings: Settings) {
 // Imported from dashboard/config/settings.js
 // Setting IDs
 export const SETTING = { PL_RANCHER_VALUE: 'rancher' };
+const REGISTRY_PATH_PROFILE = ['SOFTWARE', 'Rancher Desktop', 'Profile'];
 
 /**
- * Read and validate the json formatted deployment profile.
- * Returns the system profile if it exists, otherwise the user profile if it exists.
- * @returns A settings profile stripped of all fields not found in the schema.
+ * Read and validate deployment profiles, giving system level profiles
+ * priority over user level profiles.  If the system directory contains a
+ * defaults or locked profile, the user directory will not be read.
+ * @returns undefined, or type validated defaults and locked deployment profiles.
  */
-export function readDeploymentProfile() {
-  let profileData;
-  let profile;
+export function readDeploymentProfiles() {
+  let defaults;
+  let locked;
 
   switch (os.platform()) {
   case 'win32':
-    try {
-      profile = readRegistryDeploymentProfile(); // TODO: call with path to system profile
-    } catch {
-      try {
-        profile = readRegistryDeploymentProfile(); // TODO: call with path to user profile
-      } catch {}
+    for (const key of [nativeReg.HKLM, nativeReg.HKCU]) {
+      const registryKey = nativeReg.openKey(key, REGISTRY_PATH_PROFILE.join('\\'), nativeReg.Access.READ);
+
+      if (registryKey !== null) {
+        defaults = readRegistryUsingSchema(null, defaultSettings, registryKey, ['Defaults']);
+        locked = readRegistryUsingSchema(null, defaultSettings, registryKey, ['Locked']);
+        nativeReg.closeKey(registryKey);
+        if (typeof defaults !== 'undefined' || typeof locked !== 'undefined') {
+          break;
+        }
+      }
     }
     break;
   case 'linux':
-    try {
-      profileData = fs.readFileSync(join(paths.deploymentProfileSystem, PROFILE_JSON), 'utf8');
-    } catch {
+    for (const rootPath of [paths.deploymentProfileSystem, paths.deploymentProfileUser]) {
       try {
-        profileData = fs.readFileSync(join(paths.deploymentProfileUser, PROFILE_JSON), 'utf8');
+        const defaultsData = fs.readFileSync(join(rootPath, 'defaults.json'), 'utf8');
+
+        defaults = JSON.parse(defaultsData);
       } catch {}
-    }
-    if (typeof profileData !== 'undefined') {
-      profile = JSON.parse(profileData);
+      try {
+        const lockedData = fs.readFileSync(join(rootPath, 'locked.json'), 'utf8');
+
+        locked = JSON.parse(lockedData);
+      } catch {}
+
+      if (typeof defaults !== 'undefined' || typeof locked !== 'undefined') {
```

Comment:
Now if we set `locked` to `{}` when there's no `locked` file, this test should be an `&&`, not a `||`, assuming again that a bad `locked` profile is an error.

Reply:
unchanged, since we will now return if either object is found.